api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -625,14 +595,39 @@ private void logAdminAction(
     adminActionHistoryDao.save(adminActionHistory);
   }
 
-  /** Find users matching the user's name or email */
+  /**
+   * Find users with Registered Tier access whose name or username match the supplied search terms.
+   *
+   * @param term User-supplied search term
+   * @param sort Option(s) for ordering query results
+   * @return the List of DbUsers which meet the search and access requirements
+   * @deprecated use {@link #findUsersBySearchString(java.lang.String, java.lang.String,
+   *     org.springframework.data.domain.Sort)} instead.
+   */
+  @Deprecated
   @Override
   public List<DbUser> findUsersBySearchString(String term, Sort sort) {
-    List<Short> dataAccessLevels =
-        Stream.of(DataAccessLevel.REGISTERED, DataAccessLevel.PROTECTED)
-            .map(DbStorageEnums::dataAccessLevelToStorage)
-            .collect(Collectors.toList());
-    return userDao.findUsersByDataAccessLevelsAndSearchString(dataAccessLevels, term, sort);
+    return findUsersBySearchString(accessTierService.REGISTERED_TIER_SHORT_NAME, term, sort);
+  }
+
+  /**
+   * Find users whose name or username match the supplied search terms and who have the appropriate
+   * access tier.
+   *
+   * @param accessTierShortName the shortName of the access tier to check
+   * @param term User-supplied search term
+   * @param sort Option(s) for ordering query results
+   * @return the List of DbUsers which meet the search and access requirements
+   */
+  @Override
+  public List<DbUser> findUsersBySearchString(String accessTierShortName, String term, Sort sort) {
+    return userDao.findUsersBySearchString(term, sort).stream()
+        .filter(
+            user ->
+                accessTierService
```

Comment:
This seems potentially pretty slow, since we'll be serially making calls to the DB to lookup access tiers for each of these users. Is this something that should either be eagerly loaded onto the DbUser object (at least, for this dao access) , or else can this criteria be pushed down into the DAO query?

Reply:
updated